components/brave_perf_predictor/browser/bandwidth_savings_predictor.h

```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_PERF_PREDICTOR_BROWSER_BANDWIDTH_SAVINGS_PREDICTOR_H_
+#define BRAVE_COMPONENTS_BRAVE_PERF_PREDICTOR_BROWSER_BANDWIDTH_SAVINGS_PREDICTOR_H_
+
+#include <string>
+#include <unordered_map>
+
+#include "brave/components/brave_perf_predictor/browser/third_party_extractor.h"
+#include "components/page_load_metrics/common/page_load_metrics.mojom.h"
+#include "content/public/common/resource_load_info.mojom.h"
+
+namespace brave_perf_predictor {
+
+class BandwidthSavingsPredictor {
+ public:
+  explicit BandwidthSavingsPredictor(
+    ThirdPartyExtractor* third_party_extractor);
+  ~BandwidthSavingsPredictor();
+
+  void OnPageLoadTimingUpdated(
+    const page_load_metrics::mojom::PageLoadTiming& timing);
+  void OnSubresourceBlocked(const std::string& resource_url);
+  void OnResourceLoadComplete(const GURL& main_frame_url,
+    const content::mojom::ResourceLoadInfo& resource_load_info);
+  double predict();
+  void Reset();
+
+ private:
+  ThirdPartyExtractor* third_party_extractor_ = nullptr;
```

Comment:
these should be commented as `ThirdPartyExtractor* third_party_extractor_ = nullptr;  // not owned` - this is a chromium convention

Reply:
actually does it need a reference since this is a singleton? I'm fine either way, but might be easier to just call `ThirdPartyExtractor::GetInstance()` when you need it